ostelco-ios-client/ViewControllers/Home/SetupApplePayViewController.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  SetupApplePayViewController.swift
+//  ostelco-ios-client
+//
+//  Created by Prasanth Ullattil on 05/09/2019.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+import OstelcoStyles
+import PassKit
+import PromiseKit
+import Stripe
+import UIKit
+
+class SetupApplePayViewController: ApplePayViewController {
+
+    @IBOutlet private var buttonContainer: UIView!
+    @IBOutlet private var explanatoryCopyLabel: BodyTextLabel!
+
+    var paymentButton: PKPaymentButton?
+    var membership: Product?
+
+    lazy var linkableCopy: LinkableText = {
+        return LinkableText(
+            fullText: NSLocalizedString("For now, we only accept payment through  Visa, Mastercard and American Express through Apple Pay.\n\nPlease click on the button below to set it up.\n\nRead about our current prices", comment: "Explanation for why a user need to setup Apple Pay"),
+            linkedBits: [
+                Link(
+                    NSLocalizedString("Read about our current prices", comment: "Explanation for why a user need to setup Apple Pay: linkable part: current prices"),
+                    url: ExternalLink.currentPricing.url
+                ),
+            ])!
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        self.explanatoryCopyLabel.tapDelegate = self
+        self.explanatoryCopyLabel.setLinkableText(self.linkableCopy)
+
+        setupPaymentButton()
+    }
+
+    func setupPaymentButton() {
+        // Log information about the current Apple Pay setup.
+        let applePayError: ApplePayError? = canMakePayments()
+        switch applePayError {
+        case .unsupportedDevice?:
+            debugPrint("Apple Pay is not supported on this device")
```

Comment:
Do we need to do something during these errors?

Reply:
No, I will remove this. The object creating this should check for these things...

Reply:
Removed